Step1

graytraces opened the pull request:
step1 제출합니다.

- 질문 : view에서 받아온 param을 controller 역할을 하는 클래스에서 검증을 한 다음 사용을 하는데요, 검증함수가 private라서 junit 테스트를 돌릴수 없었습니다. 그래서 public 로 변경했는데요, 이 경우 어떻게 하는게 베스트일까요

graytraces committed:
Add exception cases and handling.

graytraces committed:
Add input checker method.

graytraces committed:
Remove all 'if' to use interface and map.

graytraces committed:
Finish first version

graytraces committed:
Add space after each rounds.

It it easier to read then before.

javajigi commented on src/main/java/calculator/Calculator.java:
```diff
@@ -1,33 +1,66 @@
+        calculatorMap.put("-", new ArithmeticMinus());
+        calculatorMap.put("*", new ArithmeticMultiple());
+        calculatorMap.put("/", new ArithmeticDivide());
+        return calculatorMap;
```
계산기 구현 👍

javajigi commented on src/main/java/racingcar/PlayRounds.java:
```diff
@@ -0,0 +1,26 @@
+        }
+    }
+
+    public ArrayList<PlayerCar> getPayerCarArrayList() {
```
ArrayList보다는 List로 반환하는 것을 추천
왜 그런지는 한번 생각해 보면 좋겠어요.

javajigi commented on src/main/java/racingcar/PlayRounds.java:
```diff
@@ -0,0 +1,26 @@
+import java.util.Random;
+
+public class PlayRounds {
+    private ArrayList<PlayerCar> payerCarArrayList;
```
변수명과 메소드 이름에 List와 같은 자료구조 이름을 사용하는 것을 추천하지 않는다.
List와 같은 복수의 데이터의 경우 복수형을 쓰는 것이 더 적합하다.

javajigi commented on src/main/java/racingcar/PlayRounds.java:
```diff
@@ -0,0 +1,26 @@
+
+    public void playSingleRounds() {
+        Random random = new Random();
+        for (int i = 0; i < payerCarArrayList.size(); i++) {
```
for(PlayerCar car: payerCarArrayList) {
}

보통 foreach 구문이라고 하는데 이 구문을 쓰면 좀 더 깔끔하게 구현 가능함.

javajigi commented on src/main/java/racingcar/PlayerCar.java:
```diff
@@ -0,0 +1,18 @@
+    private String position = "-";
+
+    public void moveCar(int condition){
+        if(condition >= 4)
```
값을 하드코딩하지 마라.
상수(static final)를 만들고 이름을 부여해 이 변수의 역할이 무엇인지 의도를 드러내라.

javajigi commented on src/main/java/racingcar/PlayerCar.java:
```diff
@@ -0,0 +1,18 @@
+
+    public void printCarPosition(){
+        System.out.println(position);
+    }
```
View 관련 로직이 같이 있는 것은 추천하지 않음.

javajigi commented on src/main/java/racingcar/RacingCar.java:
```diff
@@ -0,0 +1,46 @@
+        isValidParameter(userInput.getCarCount(), userInput.getRoundCount());
+
+        int carCountNumber = Integer.parseInt(userInput.getCarCount());
+        int roundCountNumber = Integer.parseInt(userInput.getRoundCount());
```
유효성 체크와 숫자 변환을 UserInput 생성자에서 하면 어떨까?

javajigi reviewed (changes requested):
전체적인 클래스 구조는 좋네요. 클래스 구조보다는 작은 부분에 대해 피드백 남겼어요. 테스트 코드 구현 👍

javajigi commented:
private 메소드를 테스트하는 경우 public보다는 default(접근 제어자를 쓰지 않는) 쓰면 패키지 레벨에서만 접근 가능함다.

graytraces committed:
리뷰 남겨주신것 기반으로 수정하였습니다.

1. ArrayList를 제거하였습니다. 구체적 타입 보다는 상위타입이 범용성이 높아서인것 같습니다.
2. 변수명에서 자료구조 이름을 제거했습니다.
3. foreach를 사용하였습니다.
4. 하드코딩 대신에 static final을 사용하였습니다.
5. view관련 로직은 제거하였습니다.
6. UserInput 생성자에서 형변환과 유효성검사를 하였습니다.
